### ISSUE TITLE
Undo doesn't restore systems per page

### ISSUE THREAD

**Issue by wizofaus** [musescore/MuseScore#23406]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

After certain operations that decrease the number of systems per page, undo does not restore the score to its previous state  

### Steps to reproduce

1. Create a new string orchestra score, that by default shows 3 systems per page
2. Add a note to one of the middle staves (e.g. viola) and set its pitch high or low enough that the 3rd system is forced onto the next page.
3. Undo
Expected: note should be restored to original pitch and number of systems per page restored. Actual: note is restored, but systems per page remains at 2.

In this case, Undo again (which undoes addition of the note, i.e. removes it) does actually restore the systems per page but this isn't true as a general rule (e.g. if the note was already existing when the score was loaded).
 

### MuseScore Version

4.3.1

### Regression

I don't know

### Operating system

Windows 11

**Comment by bkunda** [comment]
@DmitryArefiev pls check

**Comment by DmitryArefiev** [comment]
> 1. Create a new string orchestra score, that by default shows 3 systems per page

@wizofaus String orchestra template has 3 systems per page by default. Do you mean String quartet?

> 2. Add a note to one of the middle staves (e.g. viola) and set its pitch high or low enough that the 3rd system is forced onto the next page.

With String quartet I can't reproduce with only one note shifting


Please correct the steps or add the video

**Comment by DmitryArefiev** [comment]
The issue could be in automatic re-layout (not Undo itself). On video, when removing the 2nd note:

https://github.com/musescore/MuseScore/assets/90187801/70953a83-c47a-4402-8596-08d05103adaf

**Comment by wizofaus** [comment]
No, I mean string orchestra. By default for me it's *3* systems per page. If I add a note and drag it high (or low) enough, it re-formats to 2 systems on the first page. Undo restores the note position but not the layout.

**Comment by wizofaus** [comment]
Your video demonstrates what I saw (where you've moved the note back to its original position but the # of systems per page doesn't reset), however I wouldn't necessarily consider that a bug, as you haven't done "undo", which is explicitly a request to say "revert that last change I made and go back to the previous state".

**Comment by MarcSabatella** [comment]
Note the defaults depend on where in the world you are located - A4 vs Letter page size.  For me it was only two systems by default on Letter but changed to three when I set page size to A4.  And I could then reproduce the problem with the original steps

**Comment by wizofaus** [comment]
Sure, that might be possible, but the problem can happen in principle with any number of systems per page.

**Comment by DmitryArefiev** [comment]
> however I wouldn't necessarily consider that a bug, as you haven't done "undo", which is explicitly a request to say "revert that last change I made and go back to the previous state".

I think this is the same action (Undo and moving the note to previous place)

**Comment by DmitryArefiev** [comment]
> Note the defaults depend on where in the world you are located - A4 vs Letter page size. For me it was only two systems by default on Letter but changed to three when I set page size to A4. And I could then reproduce the problem with the original steps

Yeah, you're absolutely right. Thanks!

**Comment by DmitryArefiev** [comment]
I'll modify the summary and description

**Comment by wizofaus** [comment]
> > however I wouldn't necessarily consider that a bug, as you haven't done "undo", which is explicitly a request to say "revert that last change I made and go back to the previous state".
> 
> I think this is the same action (Undo and moving the note to previous place)

There are lots of cases where performing one action then its logical reverse (e.g. changing the duration of a note to make it longer, then explicitly changing back to previous duration) won't necessarily return the score to the same state. Whether it should always dynamically recalculate systems per page every time a note is changed in pitch is at least debatable. But "undo" always should return you to the previous state.

**Comment by MarcSabatella** [comment]
Duration isn’t layout - changing duration modifies the content of the score.  For layout, I don’t think it’s debatable at all.  The layout should be deterministic - returning a note to its original position should return the layout to its original configuration as well, regardless of which command got it there.  After all, there won’t be a command history saved with the score, and it’s important the layout after save/reload not change.

But there have long been cases where this is broken in cases that don[t involve undo.  I know I submitted such an issue against MU3 some years ago but the issue tracker is impossible to search now, so I can’t find a link.
